File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItOrderPriceTypeParcelable.java
package it.trade.android.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import it.trade.android.sdk.enums.TradeItOrderPriceType;

public class TradeItOrderPriceTypeParcelable implements Parcelable {
    private TradeItOrderPriceType priceType;
    private String displayLabel;

    TradeItOrderPriceTypeParcelable(TradeItOrderPriceType priceType, String displayLabel) {
        this.priceType = priceType;
        this.displayLabel = displayLabel;
    }

    public TradeItOrderPriceType getPriceType() {
        return priceType;
    }

    public void setPriceType(TradeItOrderPriceType priceType) {
        this.priceType = priceType;
    }

    public String getDisplayLabel() {
        return displayLabel;
    }

    public void setDisplayLabel(String displayLabel) {
        this.displayLabel = displayLabel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItOrderPriceTypeParcelable that = (TradeItOrderPriceTypeParcelable) o;

        if (priceType != that.priceType) return false;
        return displayLabel != null ? displayLabel.equals(that.displayLabel) : that.displayLabel == null;
    }

    @Override
    public int hashCode() {
        int result = priceType != null ? priceType.hashCode() : 0;
        result = 31 * result + (displayLabel != null ? displayLabel.hashCode() : 0);
        return result;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.priceType == null ? -1 : this.priceType.ordinal());
        dest.writeString(this.displayLabel);
    }

    protected TradeItOrderPriceTypeParcelable(Parcel in) {
        int tmpPriceType = in.readInt();
        this.priceType = tmpPriceType == -1 ? null : TradeItOrderPriceType.values()[tmpPriceType];
        this.displayLabel = in.readString();
    }

    public static final Parcelable.Creator<TradeItOrderPriceTypeParcelable> CREATOR = new Parcelable.Creator<TradeItOrderPriceTypeParcelable>() {
        @Override
        public TradeItOrderPriceTypeParcelable createFromParcel(Parcel source) {
            return new TradeItOrderPriceTypeParcelable(source);
        }

        @Override
        public TradeItOrderPriceTypeParcelable[] newArray(int size) {
            return new TradeItOrderPriceTypeParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/TradeItOrderPriceTypeParcelable.kt
package it.trade.android.sdk.model

import android.os.Parcel
import android.os.Parcelable

import it.trade.android.sdk.enums.TradeItOrderPriceType
import it.trade.android.sdk.enums.TradeItOrderQuantityType

class TradeItOrderPriceTypeParcelable : Parcelable {
    var priceType: TradeItOrderPriceType
    var displayLabel: String
    var supportedOrderQuantityTypes: List<TradeItOrderQuantityType> = arrayListOf()

    internal constructor(
        priceType: TradeItOrderPriceType,
        displayLabel: String,
        supportedOrderQuantityTypes: List<TradeItOrderQuantityType>
    ) {
        this.priceType = priceType
        this.displayLabel = displayLabel
        this.supportedOrderQuantityTypes = supportedOrderQuantityTypes
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TradeItOrderPriceTypeParcelable

        if (priceType != other.priceType) return false
        if (displayLabel != other.displayLabel) return false
        if (supportedOrderQuantityTypes != other.supportedOrderQuantityTypes) return false

        return true
    }

    override fun hashCode(): Int {
        var result = priceType.hashCode()
        result = 31 * result + displayLabel.hashCode()
        result = 31 * result + supportedOrderQuantityTypes.hashCode()
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.priceType.ordinal)
        dest.writeString(this.displayLabel)
        dest.writeList(this.supportedOrderQuantityTypes)
    }

    protected constructor(`in`: Parcel) {
        val tmpPriceType = `in`.readInt()
        this.priceType = TradeItOrderPriceType.values()[tmpPriceType]
        this.displayLabel = `in`.readString()
        this.supportedOrderQuantityTypes = ArrayList()
        `in`.readList(this.supportedOrderQuantityTypes, TradeItOrderQuantityType::class.java.getClassLoader())
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<TradeItOrderPriceTypeParcelable> = object : Parcelable.Creator<TradeItOrderPriceTypeParcelable> {
            override fun createFromParcel(source: Parcel): TradeItOrderPriceTypeParcelable {
                return TradeItOrderPriceTypeParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItOrderPriceTypeParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
